### DISCUSSION TITLE
Add {player} variable to /cmi scan

### DISCUSSION THREAD

**Discussion by TomLewis** [Zrips/CMI#4548] [Feature Requests]
Ahoy! I want to utilize eventCommands.yml to scan a users Echest/Inventory when they log in for a specific set of items.

You have /cmi scan but no way to target specific players, which I could then use when they log on and do a scan then.

Adding this simple feature would be super powerful with eventCommands, being able to scan a player for any event and alert staff/log it somewhere.

**Comment by Zrips** [comment]
Are we talking about scan or search command here? As those different ones and scan doesn't check enderchest while search does.

**Comment by TomLewis** [comment]
oh my god I didn't know search existed! You have so many things not on the docs! So is search for player data? and scan for World data like chests? Does search scan enderchests & inv?

**Comment by Zrips** [comment]
Search is for player data files and scan is for world files. So search will check players inventories, enderchest and could check for some specific player related things, like players game mode, so in general you can find everyone who has creative mode enabled or similar. Wiki is lacking and i need to update that one...

**Comment by TomLewis** [comment]
So you already have what i'm after! I think because I only saw the scan page on zrips I thought that was it! 
So I should be able to setup a Search for a specific player when they log in then to search for specific items, or is it a global Search?

**Comment by Zrips** [comment]
Currently its a global search. Its extremely old feature i created in recode is needed. Its mainly to find some items players could be in possession which should not be the case. It wasn't made to check one specific player because you can already do that with /cmi inv command and similar, but you have some what special case here. You want to check logging in player for specific items and i suspect you want to inform staff or something. But then again, search command works on offline players, so there is no need to wait until they join, you can just run it at any time and check if anyone have items you are looking for

**Comment by TomLewis** [comment]
Yeah I was a little bit worried about running it on 285110 players. I'm not sure if you load it all into memory first or what!

**Comment by Zrips** [comment]
Well it does try to unload player data after checking it. But i never run it on 285110 user files, so hard to say how it would handle so many of those in one go. Could be interesting thing to test and a nice benchmark for future reference while doing some improvements on this system

**Comment by TomLewis** [comment]
I will do it on my test server haha! What files does it scan? the playerdata ones in the world folder?
